client/replays/browse-local-replays.tsx

```diff
@@ -19,6 +42,273 @@ async function getReplayFolderPath() {
+  height: 32px;
+`
+
+const StyledRaceIcon = styled(RaceIcon)`
+  width: auto;
+  height: 100%;
+`
+
+const PlayerName = styled.div`
+  ${headline6};
+  ${singleLine};
+  margin-left: 16px;
+  margin-right: 8px;
+  flex-grow: 1;
+`
+
+const ReplayInfoContainer = styled.div`
+  grid-column: 6 / 9;
+  height: auto;
+
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: flex-start;
+`
+
+const StyledMapThumbnail = styled(MapThumbnail)`
+  ${shadow2dp};
+`
+
+const MapName = styled.div`
+  ${headline6};
+  ${singleLine};
+  margin-top: 8px;
+`
+
+const StyledTooltip = styled(Tooltip)`
+  width: 100%;
+  display: flex;
+  justify-content: center;
+`
+
+const ReplayInfoText = styled.div`
+  ${subtitle1};
+  ${singleLine};
+  margin: 8px 0;
+
+  color: ${colorTextSecondary};
+`
+
+const MapNoImageContainer = styled.div`
+  position: relative;
+  width: 100%;
+  height: auto;
+  border-radius: 2px;
+  contain: content;
+`
+
+const ReplayActionsContainer = styled.div`
+  display: flex;
+  justify-content: flex-end;
+  align-items: center;
+  width: 100%;
+  height: 56px;
+`
+
+export function ReplayExpansionPanel({ file }: ExpansionPanelProps) {
+  const dispatch = useAppDispatch()
+  const [replayData, setReplayData] = useState<{
+    headerData?: ReplayHeader
+    shieldBatteryData?: ReplayShieldBatteryData
+  }>()
+  const gameId = replayData?.shieldBatteryData?.gameId
+  const replayUserIds = replayData?.shieldBatteryData?.userIds
+  const gameInfo = useAppSelector(s => s.games.byId.get(gameId ?? ''))
+  const mapInfo = useAppSelector(s => s.maps2.byId.get(gameInfo?.mapId ?? ''))
+  const usersById = useAppSelector(s => s.users.byId)
+
+  const [mapNameRef, isMapNameOverflowing] = useOverflowingElement()
+  const [gameTypeRef, isGameTypeOverflowing] = useOverflowingElement()
+
+  const [parseError, setParseError] = useState(null)
+
+  useEffect(() => {
+    getReplayData(file.path)
+      .then(data => {
+        setParseError(null)
+        setReplayData(data)
+      })
+      .catch(err => setParseError(err))
+  }, [file.path])
+
+  useEffect(() => {
+    const getGameInfoAbortController = new AbortController()
+
+    if (gameId) {
+      dispatch(
+        // In case this returns an error, we just default to showing the data from the replay
+        // itself. That's why we're not utilizing the `onSuccess`/`onError` handlers here.
+        viewGame(gameId, {
+          signal: getGameInfoAbortController.signal,
+          onSuccess: () => {},
+          onError: () => {},
+        }),
+      )
+    }
+
+    return () => {
+      getGameInfoAbortController.abort()
+    }
+  }, [gameId, dispatch])
+
+  const [durationStr, gameTypeLabel, mapName, playerListItems] = useMemo(() => {
+    const replayHeader = replayData?.headerData
+    if (!replayHeader) {
+      return ['00:00', 'Unknown', 'Unknown map', null]
+    }
+
+    // TODO(2Pac): Handle replays not played at the fastest speed (if that's even possible?)
```

Comment:
This is not a thing

Reply:
Oh :d
I think I got confused by the description of this field in the `jssuh` repo, which says:
```
The duration of the game, in frames (1 / 24th of a second on fastest speed).
```
Still not fully sure what that means tbh :d

Reply:
It means what it says, your calculation is fine. What I'm saying is that "speed it was played on" is not a thing, nothing in the replay mentions gameplay speed at all, and the idea that a game was even played on a single speed is fairly erroneous (in non-network games you can change the game speed on the fly). Therefore, this TODO is not possible to achieve and it shouldn't be here